src/main/java/com/internet/shop/service/impl/UserServiceImpl.java

```diff
@@ -3,19 +3,26 @@
 import com.internet.shop.dao.UserDao;
 import com.internet.shop.lib.Inject;
 import com.internet.shop.lib.Service;
+import com.internet.shop.model.ShoppingCart;
 import com.internet.shop.model.User;
+import com.internet.shop.service.ShoppingCartService;
 import com.internet.shop.service.UserService;
 import java.util.List;
 import java.util.Optional;
 
 @Service
 public class UserServiceImpl implements UserService {
+    @Inject
+    private ShoppingCartService cartService;
+
     @Inject
     private UserDao userDao;
 
     @Override
     public User create(User user) {
-        return userDao.create(user);
+        User userDB = userDao.create(user);
+        cartService.create(new ShoppingCart(userDB.getId()));
+        return userDB;
```

Comment:
Let's cart creating and coupling  to controller. Admin user for example may not have cart

Reply:
done